tests/vite-app/cypress/integration/unittests.spec.ts

```diff
@@ -0,0 +1,131 @@
+/// <reference types="cypress" />
+import { JwtRsaVerifier } from "aws-jwt-verify";
+import {
+  ISSUER,
+  AUDIENCE,
+  JWKSURI,
+  VALID_TOKEN,
+  EXPIRED_TOKEN,
+  NOT_YET_VALID_TOKEN,
+} from "../fixtures/token-data.json";
+
+describe("unit tests", () => {
+  const INVAILD_ISSUER = "https://example.org";
+  const INVAILD_JWKSURI = "/notexample-JWKS.json";
+  const INVAILD_AUDIENCE = "notaudience";
+
+  beforeEach(() => {
+    cy.intercept("GET", JWKSURI, { fixture: "example-JWKS" });
+  });
+
+  it("valid token", async () => {
+    const verifier = JwtRsaVerifier.create({
+      issuer: ISSUER,
+      audience: AUDIENCE,
+      jwksUri: JWKSURI,
+    });
+    const payload = await verifier.verify(VALID_TOKEN);
+
+    expect(payload).to.exist;
+  });
+
+  it("expired token", async () => {
+    const verifier = JwtRsaVerifier.create({
+      issuer: ISSUER,
+      audience: AUDIENCE,
+      jwksUri: JWKSURI,
+    });
+
+    try {
+      const payload = await verifier.verify(EXPIRED_TOKEN);
+
+      expect(payload).to.not.exist;
+    } catch (ex) {
+      expect(ex.message).to.include("Token expired at ");
```

Comment:
nit: You could something like expect to throw. https://docs.cypress.io/guides/references/assertions#BDD-Assertions

Reply:
@hakanson

Reply:
The method under test is an async function, so looking to see if cypress supports [chai-as-promised](https://www.chaijs.com/plugins/chai-as-promised/) assertion language. (e.g. `promise.should.be.rejectedWith(Error)`)

Either way, it still might be better to check the Error type than the message

Reply:
no luck getting chai-as-promised working, so added checks like `expect(ex).to.be.an.instanceof(JwtExpiredError)` on the unit tests

Reply:
no worries, is still valid

Reply:
To get this working, I had to `import { JwtExpiredError } from "aws-jwt-verify/error"` - but now I am wondering with we want to use `index.ts` for barrel imports from `error.ts`?  Something like `export * from "./error"` in that file so that  `import { JwtRsaVerifier, JwtExpiredError } from "aws-jwt-verify" works as expected.

But, anything we add to `index.ts` becomes part of the API contract

Reply:
This is a design choice we made (and you can surely have opinions about):

The main `index.ts` file has just 2 exports, so the namespace is nicely small, which is nice for regular usage - because IDEs will not swamp users with symbols when they enable code completion on the import.

Errors, but also other things you might want to refer to such as from `jwt-model`or `jwk` must be imported as you did from the subpackage.

`import { JwtExpiredError } from "aws-jwt-verify/error"`
`import { JwtPayload } from "aws-jwt-verify/jwt-model"`

We like it like that. Let me rephrase, I like that :) (I think, I went through lengths to get this to work nicely, that's why the package.json has the `exports` object filled with all the submodules, to support this)

Whether you like it too or not -- it __is__ documented explicitly: https://github.com/awslabs/aws-jwt-verify#verification-errors